test/integration/backward_compatible/parallel/serialization/compact/CompactSerializersLiveTest.js

```diff
@@ -0,0 +1,194 @@
+        let cluster;
+        let client;
+        let mapName;
+        const COMPACT_ENABLED_ZERO_CONFIG_XML = `
+            <hazelcast xmlns="http://www.hazelcast.com/schema/config"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://www.hazelcast.com/schema/config
+                http://www.hazelcast.com/schema/config/hazelcast-config-5.0.xsd">
+                <network>
+                    <port>0</port>
+                </network>
+                <serialization>
+                    <compact-serialization enabled="true" />
+                </serialization>
+            </hazelcast>
+        `;
+
+        before(async function () {
+            cluster = await testFactory.createClusterForParallelTests(undefined, COMPACT_ENABLED_ZERO_CONFIG_XML);
+            const member = await RC.startMember(cluster.id);
+            client = await testFactory.newHazelcastClientForParallelTests({
+                clusterName: cluster.id,
+                serialization: {
+                    compact: {
+                        serializers: [new compactUtil.EmployeeDTOSerializer()]
+                    }
+                }
+            }, member);
+            mapName = TestUtil.randomString(10);
+        });
+
+        after(async function () {
+            await testFactory.shutdownAll();
+        });
+
+        it('should read server side data', async function() {
+            const expectedAge = 23;
+            const expectedId = 456;
+
+            const script = `
+                var EmployeeDTO = Java.type('example.serialization.EmployeeDTO');
+                var map = instance_0.getMap("${mapName}");
+                map.set(1.0, new EmployeeDTO(${expectedAge}, ${expectedId}));
+            `;
+            await RC.executeOnController(cluster.id, script, Lang.JAVASCRIPT);
+            const map = await client.getMap(mapName);
+            const value = await map.get(1);
+            value.should.be.instanceof(compactUtil.EmployeeDTO);
+            value.age.should.be.equal(expectedAge);
+            value.id.equals(expectedId).should.be.true;
+        });
+
+        it('should write correct data', async function() {
+            const expectedAge = 23;
+            const expectedId = Long.fromNumber(456);
+
+            const map = await client.getMap(mapName);
+            await map.set(1, new compactUtil.EmployeeDTO(expectedAge, expectedId));
+
+            const script = `
+                var map = instance_0.getMap("${mapName}");
+                var value = map.get(1.0);
+                result = value.class.toString() + value.age.toString() + value.id.toString();
+            `;
+            const response = await RC.executeOnController(cluster.id, script, Lang.JAVASCRIPT);
+
+            const resultString = response.result.toString();
+            // due to class loader, server gets the class object itself not generic record
```

Comment:
is the class special attribute defined by the scripting engine? I can't see that in the class itself in the Java. But anyways, I think you could use `value.getClass().getName()` here instead